rednotebook/gui/browser.py

```diff
@@ -65,12 +65,13 @@ def set_font_size(self, size):
             zoom *= 0.90
             self.set_zoom_level(zoom)
 
-        def highlight(self, search_text):
+        def highlight(self, search_words):
             # Tell the webview which text to highlight after the html is loaded
-            self.search_text = search_text
-            self.get_find_controller().search(
-                self.search_text, WebKit2.FindOptions.CASE_INSENSITIVE, MAX_HITS
-            )
+            self.search_words = search_words
+            for query in search_words:
+                self.get_find_controller().search(
+                    query, WebKit2.FindOptions.CASE_INSENSITIVE, MAX_HITS
+                )
```

Comment:
Can we stop after the first word is found?

Reply:
A little confused here since [the docs](https://lazka.github.io/pgi-docs/WebKit2-4.0/classes/FindController.html#WebKit2.FindController.search) say that the `search` method performs the search asynchronously. If so, would this be possible?